test/jdk/java/awt/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,333 @@
+                System.out.println("ComponentMoved: " + e.getSource());
+                componentMoved = true;
+            }
+
+            @Override
+            public void componentHidden(ComponentEvent e) {
+                System.out.println("ComponentHidden: " + e.getSource());
+                componentHidden = true;
+            }
+        };
+
+    private static void initializeGUI() {
+        frame = new Frame("Component Event Test");
+        frame.setLayout(new FlowLayout());
+
+        Panel panel = new Panel();
+        Button button = new Button("Button");
+        Label label = new Label("Label");
+        List list = new List();
+        list.add("One");
+        list.add("Two");
+        list.add("Three");
+        Choice choice = new Choice();
+        choice.add("Red");
+        choice.add("Orange");
+        choice.add("Yellow");
+        Checkbox checkbox = new Checkbox("Checkbox");
+        Scrollbar scrollbar = new Scrollbar(Scrollbar.HORIZONTAL, 0, 1, 0, 255);
+        TextField textfield = new TextField(15);
+        TextArea textarea = new TextArea(5, 15);
+
+        components = new Component[] { panel, button, label, list, choice,
+            checkbox, scrollbar, textfield, textarea, frame };
+
+        for (int i = 0; i < components.length - 1; i++) {
+            components[i].addComponentListener(componentListener);
+            frame.add(components[i]);
+        }
+        frame.addComponentListener(componentListener);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeAndWait(ComponentEventTest::initializeGUI);
+            robot.waitForIdle();
+            robot.delay(DELAY);
+
+            doTest();
+
+            System.out.println("Test PASSED");
+        } finally {
+            EventQueue.invokeAndWait(ComponentEventTest::disposeFrame);
+        }
+    }
+
+    private static void doTest()
+        throws InvocationTargetException, InterruptedException {
+        // Click on the Frame to ensure its gain Focus
+        EventQueue.invokeAndWait(() -> {
+            Point location = frame.getLocationOnScreen();
+            Dimension size = frame.getSize();
+            centerPoint = new Point(location.x + size.width / 2,
+                location.y + size.height / 2);
+        });
+
+        robot.mouseMove(centerPoint.x, centerPoint.y);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+        robot.delay(DELAY);
+
+        for (int i = 0; i < components.length; i++) {
+            for (boolean state : new boolean[] { true, false }) {
+                doTest(components[i], state);
+            }
+        }
+
+        robot.delay(DELAY);
+
+        System.out.println("Iconify frame");
```

Comment:
Similarly, iconify and deiconify could be moved into a separate method. Shorter, logically complete pieces improve the code quality and the overall readability of the code.